apps/total-typescript/src/path-to-purchase-react/pricing.tsx

```diff
@@ -82,29 +83,21 @@ export const Pricing: React.FC<React.PropsWithChildren<PricingProps>> = ({
   const {subscriber, loadingSubscriber} = useConvertkit()
   const router = useRouter()
 
-  const {data: formattedPrice, status} = useQuery<FormattedPrice>(
-    ['pricing', merchantCoupon, debouncedQuantity, productId, userId, couponId],
-    () =>
-      fetch('/api/skill/prices', {
-        method: 'POST',
-        body: JSON.stringify({
-          productId,
-          ...(merchantCoupon && {merchantCouponId: merchantCoupon.id}),
-          quantity,
-          purchases,
-          userId,
-          siteCouponId: couponId,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      })
-        .then((res) => res.json())
-        .then((formattedPrice: FormattedPrice) => {
-          addPrice(formattedPrice, productId)
-          return formattedPrice
-        }),
-  )
+  const {data: formattedPrice, status} = trpc.useQuery([
```

Comment:
Since this is the primary change to the 'Buy' component, I have to assume this is related to the issue I am seeing.

The issue being: when I try to toggle from 'Individual' to 'Team', it doesn't actually toggle and the page locks up.

When I change `quantity: debouncedQuantity` to just `quantity`, it allows me to toggle it once, but then when I click the toggle again, the page locks up.

This `trpc` endpoint is working well for the other page I put it on, so I'm having a hard time understanding what it is getting caught on here.

I guess the main different between this query and the one on the other page are the options going into it:

```javascript
    {
      productId,
      userId,
      quantity: debouncedQuantity,
      purchases: [],
    },
```

I'm running out of ideas for what the issue could be or how to debug it. Any insights @joelhooks ?

Reply:
Here is a video of the behavior.

https://user-images.githubusercontent.com/694063/208967930-288bae13-77e3-4337-a922-04bb0b35d6f3.mp4

A detail I just noticed this time is that a `/api/trpc/pricing.formatted` web request isn't showing up in the network log for that click, so it may not even be getting to call that in the first place.

Reply:
Turns out this was due to some weird rendering logic with the `addPrice` hook. Moving `addPrice` into an `onSuccess` handler on the tRPC query did the trick.

![image](https://user-images.githubusercontent.com/694063/208982445-891e1046-b285-42bc-9574-27ed24ee046b.png)